dotnet/src/SemanticKernel/CoreSkills/PlannerSkill.cs

```diff
@@ -25,19 +25,60 @@ namespace Microsoft.SemanticKernel.CoreSkills;
 public class PlannerSkill
 {
     /// <summary>
-    /// The name to use when creating semantic functions that are restricted from the PlannerSkill plans
+    /// Parameter names.
+    /// <see cref="ContextVariables"/>
     /// </summary>
-    private const string RestrictedSkillName = "PlannerSkill_Excluded";
+    public static class Parameters
+    {
+        /// <summary>
+        /// The number of buckets to create
+        /// </summary>
+        public const string BucketCount = "bucketCount";
+
+        /// <summary>
+        /// The prefix to use for the bucket labels
+        /// </summary>
+        public const string BucketLabelPrefix = "bucketLabelPrefix";
+
+        /// <summary>
+        /// The relevancy threshold when filtering registered functions.
+        /// </summary>
+        public const string RelevancyThreshold = "relevancyThreshold";
+
+        /// <summary>
+        /// The maximum number of functions to include in the plan.
+        /// </summary>
+        public const string MaxFunctions = "maxFunctions";
+
+        /// <summary>
+        /// The list of skills to exclude from the plan.
+        /// </summary>
+        public const string ExcludedSkills = "excludedSkills";
+
+        /// <summary>
+        /// The list of functions to exclude from the plan.
+        /// </summary>
+        public const string ExcludedFunctions = "excludedFunctions";
+
+        /// <summary>
+        /// The list of functions to include in the plan.
+        /// </summary>
+        public const string IncludedFunctions = "includedFunctions";
+    }
 
-    /// <summary>
-    /// the skills to exclude from the skill collection
-    /// </summary>
-    private static readonly List<string> s_excludedSkills = new() { RestrictedSkillName };
+    internal sealed class PlannerSkillConfig
+    {
+        public double RelevancyThreshold { get; set; } = 0.78;
+        public int MaxFunctions { get; set; } = 10;
+        public HashSet<string> ExcludedSkills { get; } = new() { RestrictedSkillName };
+        public HashSet<string> ExcludedFunctions { get; } = new() { "CreatePlan", "ExecutePlan" };
```

Comment:
>new() { "CreatePlan", "ExecutePlan" }; [](http://example.com/codeflow?start=60&length=38)

Can we make it so these are always excluded? I think right now if a user passes any excluded skills or functions, it will overwrite these, right?

Reply:
Yes, that is the current behavior. See `GetPlannerSkillConfig`. Will add some UTs to make this more apparent.